core/src/layers/logging.rs

```diff
@@ -108,8 +108,11 @@ impl LoggingLayer {
     /// `None` means disable the log for error.
     pub fn with_error_level(mut self, level: Option<&str>) -> Result<Self> {
         if let Some(level) = level {
-            let level = Level::from_str(level).expect("parse level must succeed");
-            self.error_level = Some(level);
+            if let Ok(level) = Level::from_str(level) {
```

Comment:
How about `let lvl = level.parse().map_err(xxx)?`? Maybe we can also put the input level into error context?